app/services/vendor_api/generate_test_applications_for_provider.rb

```diff
@@ -1,27 +1,68 @@
 module VendorAPI
   class GenerateTestApplicationsForProvider
-    def call(provider:, courses_per_application:, count:, for_ratified_courses: false)
+    def initialize(provider:, courses_per_application:, count:, for_training_courses: false, for_ratified_courses: false, for_test_provider_courses: false)
+      @provider = provider
+      @courses_per_application = courses_per_application
+      @application_count = count
+
+      if [for_training_courses, for_ratified_courses, for_test_provider_courses].none?
+        for_training_courses = true
+      end
+      @for_training_courses = for_training_courses
+      @for_ratified_courses = for_ratified_courses
+      @for_test_provider_courses = for_test_provider_courses
+    end
+
+    def call
       raise ParameterInvalid, 'Parameter is invalid (cannot be zero): courses_per_application' if courses_per_application.zero?
 
-      course_ids = course_list_for_provider(provider, for_ratified_courses).pluck(:id)
+      application_count.times do
+        course_ids = random_course_ids_to_apply_for
 
-      raise ParameterInvalid, 'Parameter is invalid (cannot be greater than number of available courses): courses_per_application' if course_ids.count < courses_per_application
+        raise ParameterInvalid, 'Parameter is invalid (cannot be greater than number of available courses): courses_per_application' if course_ids.count < courses_per_application
 
-      GenerateTestApplicationsForCourses.perform_async(course_ids, courses_per_application, count)
+        GenerateTestApplicationsForCourses.perform_async(course_ids, courses_per_application)
```

Comment:
I've been having a think around this: What if we call the test generator as many times as flags, example:
I've set all 3 flags as true. I grab the courses for each of those, and call `GenerateTestApplicationsForCourses.perform_async(course_ids, courses_per_application, count)` three time, each time with the specific course (ratified courses or training courses). This way we don't have to worry about the random sampling later on. Only thing we need to pass in then is the count which we can spread across the options, for instance the count requested is 30, I can pass 10 to each job. Happy to chat through this and hear your thoughts

Reply:
I don't think this will meet the need for mixed applications :( 
We need this to generate `count` application _forms_, with `courses_per_application` application choices, where the course choices (per application form) are spread between the course types specified.

For example, if we do `?count=2&courses_per_application=3&for_ratified_courses=true&for_training_courses=true&for_test_provider_courses=true`. Then I'd expect to get 2 application forms, each with 3 application choices. On each form, there should be an application choice applying to each of: a course run by the provider, a course ratified by the provider, and a course run by the test provider.

Reply:
bah ok makes sense, forgot about the mixed bit!